pkg/peermgr/swarm.go

```diff
@@ -213,6 +220,7 @@ func (swarm *Swarm) Ping() {
 			ticker.Stop()
 			ticker = time.NewTicker(swarm.pingTimeout)
 		case <-swarm.ctx.Done():
+			swarm.cancel()
```

Comment:
cancel() should have  been called before to trigger ctx.Done()

Reply:
remove it